### PR TITLE
Update tensorboard to 2.16.2

### PR DESCRIPTION
**Pull request by rikenm1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Update tensorboard to 2.16.2 and tensorboard-data-server to 0.8

###### Change Log  

- Change: Update tensorboard to 2.16.2
- Change: Add tensorboard as runtime req of tensorflow
- Change: Add auto upgrade script for easy tensorboard update in the future

###### Does this affect the toolchain?  

**YES/NO**
NO

###### Test Methodology

- Pipeline build id: [559826](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=559826&view=results)

### CODE REVIEW

File: SPECS/python-tensorboard/python-tensorboard.spec [hunk 1]
```diff
@@ -6,24 +6,32 @@ TensorBoard is a suite of web applications for inspecting and understanding your
...
+BuildRequires:  python3-numpy
 BuildRequires:  bazel
 BuildRequires:  gcc
 BuildRequires:  build-essential
-BuildRequires:  protobuf
+BuildRequires:  python3-protobuf
 BuildRequires:  zlib
 BuildRequires:  python3-virtualenv
+BuildRequires:  python3-werkzeug
+BuildRequires:  python3-six
```
Comment (anphel31) [inline SPECS/python-tensorboard/python-tensorboard.spec:34]: this is a duplicate

File: SPECS/python-tensorboard/python-tensorboard.spec [hunk 2]
```diff
@@ -6,24 +6,32 @@ TensorBoard is a suite of web applications for inspecting and understanding your
...
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 URL:            https://github.com/tensorflow/tensorboard
-Source0:        https://github.com/tensorflow/tensorboard/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
-Source1:        %{name}-%{version}-cache.tar.gz
+# This source also contains the dependencies required for building tensorboard
+Source0:        %{_distro_sources_url}/%{name}-%{version}.tar.gz#/%{name}-%{version}.tar.gz
+
+Patch0:         0000-Use-system-package.patch
 BuildRequires:  python3-setuptools
```
Comment (anphel31) [inline SPECS/python-tensorboard/python-tensorboard.spec:19]: could you alphabetize the list of BuildRequires? should make it easier to read

File: SPECS/python-tensorboard/python-tensorboard.spec (hunk 1 above)
Comment (rikenm1) [reply SPECS/python-tensorboard/python-tensorboard.spec]: removed

File: SPECS/python-tensorboard/python-tensorboard.spec (hunk 2 above)
Comment (rikenm1) [reply SPECS/python-tensorboard/python-tensorboard.spec]: alphabetized BRs

**Review by anphel31 — APPROVED** [review]

File: SPECS/python-tensorboard/dockerfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    pid=$! && \
+    while true; do \
+        if grep -q "Compiling" /tmp/bazel_output.log; then \
+            echo "Compiling detected, exiting..." && \
+            kill $pid && \
+            exit 0; \
+        fi; \
+        sleep 1; \
+    done
+
```
Comment (PawelWMS) [inline SPECS/python-tensorboard/dockerfile:41]: Wouldn't running only `RUN bazel --output_user_root=./tb_tmp fetch //tensorboard/pip_package:build_pip_package` and then building with `--nofetch` work? Why the additional compilation?

File: SPECS/python-tensorboard/dockerfile [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+# iii) Python version should match the version in caching phase and the build phase
+
+FROM mcr.microsoft.com/cbl-mariner/base/core:2.0
+
+ARG TB_VERSION
+
+WORKDIR /root/
+# RUN tdnf -y install createrepo
+# COPY RPMS /opt/rpms
+# COPY custom.repo /etc/yum.repos.d
```
Comment (PawelWMS) [inline SPECS/python-tensorboard/dockerfile:14]: Please clean-up unused command.s

File: SPECS/python-tensorboard/genereate_source_tarball.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+TB_VERSION="2.16.2"
+
+docker build --build-arg TB_VERSION=$TB_VERSION -t tensorboard .
+
+CONTAINER_ID=$(docker run -d tensorboard)
+
+docker cp $CONTAINER_ID:/root/tensorboard-$TB_VERSION.tar.gz $PWD
+
+docker stop $CONTAINER_ID
+docker rm $CONTAINER_ID
```
Comment (PawelWMS) [inline SPECS/python-tensorboard/genereate_source_tarball.sh:14]: I don't think this format will work with our upgrade automation. Please sync with Nicolas but I think at the minimum we need these scripts to accept the `srcTarball`, `outFolder`, and `pkgVersion` arguments. See `SPECS/etcd/generate_source_tarball.sh` for an example.

File: SPECS/python-tensorboard/dockerfile [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+# COPY toolchain_rpms /opt/rpms
+# RUN createrepo /opt/rpms
+#bazel version must match with the .bazelversion
+RUN tdnf -y install build-essential git tar bazel python3-devel python3-numpy python3-pip python3-wheel python3-six python3-setuptools python3-virtualenv python3-protobuf python3-absl-py gcc zlib wget ca-certificates
+RUN tdnf -y install which python3-werkzeug
+# Create Working Directory
+RUN wget https://github.com/tensorflow/tensorboard/archive/refs/tags/$TB_VERSION.tar.gz
```
Comment (PawelWMS) [inline SPECS/python-tensorboard/dockerfile:22]: Please split the cache and the upstream source.

**Pull request merged**